pkg/alertmanager/validation/v1alpha1/validation.go

```diff
@@ -374,6 +375,14 @@ func validateAlertManagerRoutes(r *monitoringv1alpha1.Route, receivers, muteTime
 		}
 	}
 
+	// Validate matchers, so they don't end up crashing the main instance: https://github.com/prometheus/alertmanager/blob/v0.26.0/config/coordinator.go#L124.
+	for _, matcher := range r.Matchers {
```

Comment:
yes correct.

Reply:
ACK, so it's safe to say that `v1` versions will always be available alongside `alpha` and/or `beta` (since otherwise, the implicit version upgrade won't take place, and this functionality will be left out)? I'm assuming that's the case and moving this to `v1` based on the conversation above. 👍🏼

Reply:
I see a lot of code in `alpha` and `beta` has overlapping logic as well. What if we merged the two under `v1` and used interface parameters, getting rid of them entirely?

Reply:
I'm not against at all, but I'd suggest to do this on a dedicated PR.